docs/tutorials/embassy/index.md

```diff
@@ -0,0 +1,380 @@
+
+```shell
+elf2uf-rs -d -s /path/to/your/binary
+```
+
+* `-d` to automatically deploy to a mounted pico
+* `-s` to open the pico as a serial device after deploy and print serial output
+  
+:::note
+On `Windows`, you may need to run this command in a terminal that has **Admin Privileges**.
+:::
+
+## Debugging using Raspberry Pi Debug Probe
+
+In order to be able to debug the program running on the board, we will need to connect the **Raspberry Pi Debug Probe** to our **Raspberry Pi Pico W**. Below, you have a picture of the debug kit provided:
+
+![Raspberry Pi Debug probe](assets/the-probe.png)
+
+To connect them, we will use the **3-pin debug to 0.1-inch header (female)** cable. First, carefully insert the **3-pin debug** head in the **right side** connector. Then you will also need to connect it to the Raspberry Pi Pico W. You will find attached the pinout, take a closer look at the bottom of the image:
+
+![Raspberry Pi Pico W pinout](assets/picow-pinout.svg)
+
+The connections must be:
+
+| Wire | Raspberry Pi Pico W |
+|-|-|
+|TX (Orange)|SWCLK|
+|GND (Black)|GND|
+|RX (Yellow)|SWDIO|
+
+:::warning
+Do not forget to connect both the Debug Probe and Pico to your PC.
+:::
+
+Now, you can either debug using the command line by running:
+
+```shell
+probe-rs run --chip RP2040 path/to/your/binary
+```
+
+or you can use **Debug and Run** view in Visual Studio Code. You will need to modify the `programBinary` path in the `.vscode/launch.json` config file to point to your binary file.
+
+## Building your first Embassy-rs project
+
+In this section, we will briefly go over the steps you need to take in order to get your first project using **Rust** and **Embassy-rs** going.
+
+### Creating your crate
+
+The first step is to create your cargo package by running the following command in your terminal:
+
+```shell
+cargo new --vcs none embassy
+```
+
+* `--vcs none` because at the moment we do not want to use any code versioning (they are useful, but this is not the purpose of this tutorial)
+
+### Crate settings
+
+Because we are running in an embedded environment, our code needs to be *"tailored"* specifically for the microcontroller we intend to use. In our case, it is the **RP2040**, but these general steps apply for any chip, produced by any manufacturer.
+
+#### No standard library
+
+Due to the size constraints imposed on us (in our case, `2MB` of flash memory), the standard library has to go. We specify that by adding the `#![no_std]` attribute to the beginning of our `src/mains.rs` file.
+
+#### No `main` function
+
+Because we are using the **Embassy-rs** framework, we want to let it take care of entry point of our program (because it has to do some complex operations, like allocating the `task-arena` and `executor` structures). For the moment, all we will need to do is add the `#![no_main]` attribute to `src/main.rs`.
```

Comment:
```suggestion
Because we are using the **Embassy-rs** framework, we want to let it take care of the entry point of our program (because it has to do some complex operations, like allocating the `task-arena` and `executor` structures). For the moment, all we will need to do is add the `#![no_main]` attribute to `src/main.rs`.
```